src/Jwk/JsonWebKeySet.cs

```diff
@@ -54,11 +54,19 @@ public JsonWebKeySet(string json)
 
         var jwebKeys = JsonSerializer.Deserialize<JsonWebKeySet>(json);
         Keys = jwebKeys.Keys;
+        
+        RawData = json;
     }
 
     /// <summary>
     /// A list of JSON web keys
     /// </summary>
     [JsonPropertyName("keys")]
     public List<JsonWebKey> Keys { get; set; } = new();
-}
\ No newline at end of file
+    
+    /// <summary>
+    /// The JSON string used to deserialize this object
+    /// </summary>
+    [JsonIgnore]
+    public string RawData { get; set; };
```

Comment:
I wasn't entirely sure what an appropriate name for this property should be. Thinking about it again, maybe I should have suggested to use `Raw` instead of `RawData`, in keeping with the precedent set by the `ProtocolResponse` [class](https://github.com/IdentityModel/IdentityModel/blob/d27f1236e1eb1b8d8b39cf39c6f0dd8a114a8544/src/Client/Messages/ProtocolResponse.cs#L126). And maybe the `set` should be protected or private?